aaron/scripts/articleView.js

```diff
@@ -0,0 +1,128 @@
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = () => {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = () => {
+  $('nav').on('click', '.tab', function(e) {
+    e.preventDefault();
+    $('.tab-content').hide();
+    $(`#${$(this).data('content')}`).fadeIn();
+  });
+
+  $('nav .tab:first').click();
+};
+
+articleView.setTeasers = () => {
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('article').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    if ($(this).text() === 'Read on →') {
+      $(this).parent().find('*').fadeIn();
+      $(this).html('Show Less &larr;');
+    } else {
+      $('body').animate({
+        scrollTop: ($(this).parent().offset().top)
+      },200);
+      $(this).html('Read on &rarr;');
+      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+    }
+  });
+};
+
+// COMMENT: Where is this function called? Why?
+// This function is called in the new.html file. It is called there to initialize the content on it when it loads.
+articleView.initNewArticlePage = () => {
+  // DONE: The new articles we create will be copy/pasted into our source data file.
+  // Set up this "export" functionality. We can hide it for now, and show it once we have data to export.
+
+  $('#article-json').on('focus', function(){
+    this.select();
+  });
+
+  // DONE: Add an event handler to update the preview and the export field if any inputs change.
+  $('form').on('input', () => {
+    articleView.create();
+  })
+
+};
+
+articleView.create = () => {
+
+  // DONE: Clear out the #articles element, so we can put in the updated preview
+  $('#articles').empty();
+
+  // DONE: Instantiate an article based on what's in the form fields:
+  let newArticle = new Article({
+    title: $('#title').val(),
+    author: $('#author').val(),
+    authorUrl: $('#authorUrl').val(),
+    category: $('#category').val(),
+    body: $('#body').val(),
+    publishedOn: new Date(),
+  });
+
+  // DONE: Use our interface to the Handblebars template to put this new article into the DOM:
+  $('#articles').append(newArticle.toHtml());
+
+  // DONE: Activate the highlighting of any code blocks; look at the documentation for hljs to see how to do this by placing a callback function in the .each():
+  $('pre code').each();
+
+  // DONE: Show our export field, and export the new article as JSON, so it's ready to copy/paste into blogArticles.js:
+  $('#export').val(JSON.stringify(newArticle));
+};
+
+// COMMENT: Where is this function called? Why?
+// This function is called at the bottom of new.html. It is called there so it gets loaded on just the new.html page
```

Comment:
are you sure this is being called at the bottom of new.html? It is called "initIndexPage" so that would be a terrible name if it was initializing the New page. Also, look at what it is doing, the functions it is calling are not things that happen on new.html